crypto/threads_pthread.c

```diff
@@ -76,8 +77,10 @@ int CRYPTO_THREAD_read_lock(CRYPTO_RWLOCK *lock)
     if (pthread_rwlock_rdlock(lock) != 0)
         return 0;
 # else
-    if (pthread_mutex_lock(lock) != 0)
+    if (pthread_mutex_lock(lock) != 0) {
+        ossl_assert(errno != EDEADLK && errno != EBUSY);
```

Comment:
`ossl_assert` is useful where you want to return an error in a non-debug build, and cause an assertion failure in a debug build, e.g.

````
    if (!ossl_assert(check_this_is_true))
        return 0;
````
Since there is no conditional here just `assert` is good enough. Similarly for the other instance of `ossl_assert`.

Reply:
Changed `ossl_assert` to `assert`  I guess I was mistaken to think we always wanted the former not the latter.  Thanks.